app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -95,9 +99,28 @@ class MainActivity : ComponentActivity() {
         WindowCompat.setDecorFitsSystemWindows(window, false)
 
         setContent {
-            val systemUiController = rememberSystemUiController()
+            val uiModeManager = getSystemService(Context.UI_MODE_SERVICE) as UiModeManager
             val darkTheme = shouldUseDarkTheme(uiState)
 
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                uiModeManager.setApplicationNightMode(
```

Comment:
I think we should pass through the `UiModeManager.MODE_NIGHT_AUTO` if the user preference is `DarkThemeConfig.FOLLOW_SYSTEM`.